### ISSUE TITLE
Host Json Being Ignored => Pulling multiple queue messages with batchSize : 1

### ISSUE THREAD

**Issue by rowana16** [Azure/azure-functions-host#3676]
#### Investigative information

Please provide the following:

- Timestamp:  6:22:54 EST
- Function App version (1.0 or 2.0):  2.0  App Service
- Function App name:
- Function name(s) (as appropriate): 
- Invocation ID: 0af97e7a-537e-46e1-bab0-2c229ca05205
- Region: East US2

#### Repro steps

Provide the steps required to reproduce the problem:
Function App with host.json: 
{
  "applicationInsights": {
    "sampling": {
      "isEnabled": false,
      "maxTelemetryItemsPerSecond": 50
    }
  },
  "queues": {
    "maxPollingInterval": 750,
    "batchSize": 1,
    "maxDequeueCount": 2
  },
  "version": "2.0"
}
Pulls dozens of messages from the queue instead of the expected 1.

#### Expected behavior

Provide a description of the expected behavior.
Should only pull 1 queue message.  Overloads the working memory.  Crashes.

#### Actual behavior

Provide a description of the actual behavior observed.
No account is found.

**Comment by Morcatko** [comment]
I am facing the very same issue. I am trying to process images by a Function so I need to limit number of parallel executions due to high memory usage.

My host.json
```
{
  "version": "2.0",
  "queues": {
    "visibilityTimeout": "00:03:00",
    "batchSize": 4,
    "newBatchThreshold": 2
  }
}
```

I added Environment.MachineName and current process id to logs so I think I can be sure that all messages are processed on single machine/process.

No matter what I set into host.json it always executes "random" number of messages at once (which fails in my case because of not enough memory).
When I run it locally with storage emulator it seems to process 12 messages at once while on Azure it processes 16 messages.
